exporter/sumologicexporter/sender.go

```diff
@@ -270,3 +282,125 @@ func (s *sender) batch(log pdata.LogRecord, metadata Fields) ([]pdata.LogRecord,
 func (s *sender) count() int {
 	return len(s.buffer)
 }
+
+// count returns number of metrics in buffer
+func (s *sender) countMetrics() int {
+	return len(s.metricBuffer)
+}
+
+// sendMetrics sends metrics in right format basing on the s.config.MetricFormat
+func (s *sender) sendMetrics() ([]metricPair, error) {
+	switch s.config.MetricFormat {
+	case Carbon2Format:
+		return s.sendMetricsCarbon2Format()
+	default:
+		return nil, errors.New("unexpected metric format")
+	}
+}
+
+// batchMetric adds metric to the metrics buffer and flush them if buffer is full to avoid overflow
+// returns list of metric records which were not sent successfully
+func (s *sender) batchMetric(metric metricPair) ([]metricPair, error) {
+	s.metricBuffer = append(s.metricBuffer, metric)
+
+	if s.countMetrics() == maxBufferSize {
+		dropped, err := s.sendMetrics()
+		s.cleanBuffer()
+		return dropped, err
+	}
+
+	return nil, nil
+}
+
+func (s *sender) sendMetricsCarbon2Format() ([]metricPair, error) {
+	var (
+		body strings.Builder
+		errs []error
+		// droppedRecords tracks all dropped records
+		droppedRecords []metricPair
+		// currentRecords tracks records which are being actually processed
+		currentRecords []metricPair
+		nextLines      []string
+	)
+
+	for _, record := range s.metricBuffer {
+		switch record.metric.DataType() {
+		case pdata.MetricDataTypeIntGauge:
+			for i := 0; i < record.metric.IntGauge().DataPoints().Len(); i++ {
+				dataPoint := record.metric.IntGauge().DataPoints().At(i)
+				nextLines = append(nextLines,
+					fmt.Sprintf("%s  %d %d",
+						s.filter.Carbon2TagString(record),
+						dataPoint.Value(),
+						dataPoint.Timestamp()/1e9))
```

Comment:
* I'd be inclined to make a separate method for these `Sprintf`s
* IMO the following is way better `<styleguide_fight>`

```suggestion
				nextLines = append(nextLines,
					fmt.Sprintf("%s  %d %d",
						s.filter.Carbon2TagString(record),
						dataPoint.Value(),
						dataPoint.Timestamp()/1e9,
					),
				)
```

`</styleguide_fight>`